json_feature.cpp

```diff
@@ -1,27 +1,38 @@
 
 #include "json_feature.hpp"
 
-void JSONFeature::add_properties(const osmium::OSMObject& object, const char* id_name) {
+void JSONFeature::add_properties(const osmium::OSMObject& object) {
     m_writer.String("properties");
 
     m_writer.StartObject();
 
-    m_writer.String(id_name);
-    m_writer.Int64(object.id());
+    m_writer.String((m_attr_prefix + "id").c_str());
+    m_writer.Int64(object.type() == osmium::item_type::area ? osmium::area_id_to_object_id(object.id()) : object.id());
+
+    m_writer.String((m_attr_prefix + "type").c_str());
+    if (object.type() == osmium::item_type::area) {
+        if ((object.positive_id() & 0x1) == 0) {
```

Comment:
You have to add the `const`. As the message says, you want to convert "from 'const osmium::OSMObject' to 'osmium::Area'", which would remove the `const`, and that's not allowed.